### PR TITLE
feat: add new versions/all API endpoint

### PR DESCRIPTION
**Pull request by mcdurdin**
See https://github.com/keymanapp/keyman/wiki/downloads.keyman.com-version-API#all-versions-api for documentation.

### CODE REVIEW

File: api/versioninfoall.php
Comment (mcdurdin) [inline api/versioninfoall.php:1]: This file is based on versioninfo.php, with most irrelevant details stripped out.

File: api/versioninfoall.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?php
+  /*
+    API documentation: https://github.com/keymanapp/keyman/wiki/downloads.keyman.com-version-API#all-versions-api
+
+    Returns a list of all versions of Keyman, starting with version 10.0, grouped by major version:
+      { "10.0": { "alpha": ["10.0.1", ...], "beta": ["10.0.111", ...], "stable": ["10.0.112", ...] }, "11.0": ..., ... }
+
+    Uses data from web folder as canonical across all platforms
```
Comment (darcywong00) [inline api/versioninfoall.php:8]: I suppose web is canonical from 14.0 onward.

I looked back at https://downloads.keyman.com/android/stable/ and noticed android had different versions before that.

File: api/versioninfoall.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  //
+  // Parameter checks for platforms
+  //
+
+  if(isset($_REQUEST['version'])) {
+    $version = $_REQUEST['version'];
+  } else {
+    $version = '1.0';
```
Comment (darcywong00) [inline api/versioninfoall.php:30]: Does this mean the wiki should have an optional parameter?

`version` The content format version to return. If omitted, returns version 1.0 format.

File: api/versioninfoall.php
Comment (darcywong00) [reply api/versioninfoall.php:1]: I guess it would be overkill to refactor the common bits into a version class :smile:

File: api/web.config [hunk 3]
```diff
@@ -13,6 +13,10 @@
           <match url="^version/(1.0|2.0)$" />
           <action type="Rewrite" url="versioninfo.php?version={R:1}" appendQueryString="true" />
         </rule>
+        <rule name="version/all" stopProcessing="true">
+          <match url="^version/all$" />
+          <action type="Rewrite" url="versioninfoall.php" appendQueryString="true" />
+        </rule>
```
Comment (darcywong00) [inline api/web.config:19]: Is web.config still being used? If not maybe we remove on a separate PR.

File: api/versioninfoall.php (hunk 1 above)
Comment (mcdurdin) [reply api/versioninfoall.php:8]: Good catch. I will update to that minimum for now.

File: api/versioninfoall.php (hunk 2 above)
Comment (mcdurdin) [reply api/versioninfoall.php:30]: I don't think so -- given it is only v1.0. Just leaving the infrastructure there for the future.

File: api/web.config (hunk 3 above)
Comment (mcdurdin) [reply api/web.config:19]: Still used for test dev. We don't have a dockerized version of this site.

File: api/versioninfoall.php
Comment (mcdurdin) [reply api/versioninfoall.php:1]: for now, yes. yak shave.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**